link/src/main/java/com/stripe/android/link/LinkActivity.kt

```diff
@@ -19,30 +24,47 @@ import com.stripe.android.link.ui.paymentmenthod.PaymentMethodScreen
 import com.stripe.android.link.ui.signup.SignUpScreen
 import com.stripe.android.link.ui.verification.VerificationScreen
 import com.stripe.android.link.ui.wallet.WalletScreen
+import com.stripe.android.ui.core.CircularProgressIndicator
 
-internal class LinkActivity : AppCompatActivity() {
-    @VisibleForTesting
-    internal var viewModelFactory: ViewModelProvider.Factory = LinkActivityViewModel.Factory()
-    private val viewModel: LinkActivityViewModel by viewModels { viewModelFactory }
+internal class LinkActivity : ComponentActivity() {
+    internal var viewModel: LinkActivityViewModel? = null
 
     @VisibleForTesting
     internal lateinit var navController: NavHostController
 
+    @Suppress("SwallowedException")
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
 
+        try {
+            viewModel = ViewModelProvider(this, LinkActivityViewModel.factory())[LinkActivityViewModel::class.java]
+        } catch (e: NoArgsException) {
+            setResult(Activity.RESULT_CANCELED)
```

Comment:
Why is this canceled instead of being a failure?

Reply:
I followed the pattern [here](https://github.com/stripe/stripe-android/blob/9c983c0438713d34c74052c41daa28caa7d0a53c/link/src/main/java/com/stripe/android/link/LinkForegroundActivity.kt#L82), but I'm happy to change it to a failure